crates/bevy_ecs/src/archetype.rs

```diff
@@ -45,6 +46,32 @@ pub struct AddBundle {
     pub(crate) bundle_status: Vec<ComponentStatus>,
 }
 
+pub(crate) trait BundleComponentStatus {
+    /// Returns the Bundle's component status for the given "bundle index"
+    ///
+    /// # Safety
+    /// Callers must ensure that index is always a valid bundle index for the
+    /// Bundle associated with this [`BundleComponentStatus`]
+    unsafe fn get_status(&self, index: usize) -> ComponentStatus;
+}
+
+impl BundleComponentStatus for AddBundle {
+    #[inline]
+    unsafe fn get_status(&self, index: usize) -> ComponentStatus {
+        // SAFETY: caller has ensured index is a valid bundle index for this bundle
+        *self.bundle_status.get_unchecked(index)
+    }
+}
+
+pub(crate) struct SpawnBundleStatus;
+
+impl BundleComponentStatus for SpawnBundleStatus {
+    #[inline]
+    unsafe fn get_status(&self, _index: usize) -> ComponentStatus {
+        ComponentStatus::Added
```

Comment:
```suggestion
       // Components added during a spawn_bundle call are always treated as added
        ComponentStatus::Added
```

Reply:
Good call! I applied this manually as the indentation was off in this suggestion.